lib/improver/tests/nowcasting/nowcasting/test_ApplyOrographicEnhancement.py

```diff
@@ -0,0 +1,457 @@
+        when the units of the input cube do not require conversion to mm/hr."""
+        expected0 = np.array([[[[1/32., 1/32., 0.],
+                                [1., 2., 3.],
+                                [3., 3., 4.]]]])
+        precip_cube = self.precip_cube.copy()
+        precip_cube.convert_units("mm/hr")
+        precip_cube.data = np.array([[[[-1., -1., 0.],
+                                       [1., 2., 3.],
+                                       [3., 3., 4.]]]])
+        self.precip_cube.convert_units("mm/hr")
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin.apply_minimum_precip_rate(precip_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.units, Unit("mm/hr"))
+        self.assertEqual(result.metadata, precip_cube.metadata)
+        self.assertArrayAlmostEqual(result.data, expected0)
+
+    def test_tolerance(self):
+        """Test that the minimum precipitation rate is applied correctly,
+        when the allowed tolerance is altered."""
+        expected0 = np.array([[[[1/32., 1/32., 1/32.],
+                                [1/32., 1/32., 1/32.],
+                                [1/32., 1/32., 1/32.]]]])
+        precip_cube = self.precip_cube.copy()
+        precip_cube.convert_units("mm/hr")
+        precip_cube.data = np.array([[[[-1., -1., 0.],
+                                       [1., 2., 3.],
+                                       [3., 3., 4.]]]])
+        self.precip_cube.convert_units("mm/hr")
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin.apply_minimum_precip_rate(precip_cube, tolerance=10)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.units, Unit("mm/hr"))
+        self.assertEqual(result.metadata, precip_cube.metadata)
+        self.assertArrayAlmostEqual(result.data, expected0)
+
+    def test_NaN_values(self):
+        """Test a minimum precipitation rate is applied, when NaN values are
+        present within the input."""
+        expected0 = np.array([[[[1/32., 1/32., np.NaN],
+                                [1., 2., np.NaN],
+                                [3., 3., 4.]]]])
+        precip_cube = self.precip_cube.copy()
+        precip_cube.convert_units("mm/hr")
+        precip_cube.data = np.array([[[[-1., -1., np.NaN],
+                                       [1., 2., np.NaN],
+                                       [3., 3., 4.]]]])
+        precip_cube.convert_units("m/s")
+        plugin = ApplyOrographicEnhancement("add")
+        result = plugin.apply_minimum_precip_rate(precip_cube)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.units, Unit("m/s"))
+        self.assertEqual(result.metadata, precip_cube.metadata)
+        result.convert_units("mm/hr")
+        self.assertArrayAlmostEqual(result.data, expected0)
+
+
+class Test_process(IrisTest):
```

Comment:
We really need a test for how this plugin handles masked data.  We need masked data to be ignored and the mask to be preserved on the output cube.

Reply:
I've added a couple of unit tests where masked arrays are used.